gladier/tools/builtins/choice_state.py

```diff
@@ -153,32 +153,58 @@ def flow_dict(self) -> JSONObject:
 
 
 # TODO: add support here
-class ChoiceState(GladierBaseState):
+class ChoiceState(AWSBaseState):
+    """
+    The Choice state allows for branching logic depending on a set of conditions.
+
+    An example is below:
+
+    .. code-block:: python
+
+        choice_state = (
+            ChoiceState()
+            .choice(
+                ChoiceOption(
+                    rule=ComparisonRule(
+                        Variable=random_step.path_to_return_val(), NumericEquals=0.0
```

Comment:
Might want to use a `Variable` value here that is an explict JSONPath rather than this example that depends on this other state's output in the form returned by the ComputeStep.